File: Data Structures & Algorithms/duplicate-integer/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let obj = new Set()

        for(const num of nums){
            if(obj.has(num)){
                return true
            }

            obj.add(num)
        }
        return false

    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        max_seq = 0
        obj = set(nums)

        for num in obj:
            if (num - 1) not in obj:
                current_num = num
                current_streak = 1
            
                while (current_num + 1) in obj:
                    current_num += 1
                    current_streak += 1

                max_seq = max(max_seq, current_streak)

        return max_seq
